lib/charms/alertmanager_k8s/v0/alertmanager_remote_configuration.py

```diff
@@ -0,0 +1,1890 @@
+                )
+        return config
+
+    def templates(self) -> list:
+        """Returns Alertmanager templates sent inside the relation data bag.
+
+        If the `alertmanager-remote-configuration` relation exists and the relation data bag
+        contains Alertmanager templates, returns the templates in the form of a list.
+
+        Returns:
+            list: Alertmanager templates
+        """
+        templates = []
+        remote_configuration_relation = self._charm.model.get_relation(self._relation_name)
+        if remote_configuration_relation and remote_configuration_relation.app:
+            try:
+                templates_raw = remote_configuration_relation.data[
+                    remote_configuration_relation.app
+                ]["alertmanager_templates"]
+                templates = json.loads(templates_raw)
+            except KeyError:
+                logger.warning(
+                    "Remote config provider relation exists, but no templates have been provided."
+                )
+        return templates
+
+    @staticmethod
+    def _config_is_valid(config: dict) -> bool:
+        """Validates Alertmanager configuration.
+
+        Uses JSON Schema validator to check whether the Alertmanager configuration provided
+        by the `alertmanager-remote-configuration` consumer is valid.
+
+        Args:
+            config: Alertmanager configuration JSON
+
+        Returns:
+            bool: True/False depending on the configuration validity
+        """
+        try:
+            validate(instance=config, schema=ALERTMANAGER_CONFIG_JSON_SCHEMA)
+            return True
+        except exceptions.ValidationError:
+            return False
+
+    def _get_current_config(self) -> str:
+        """Gets current configuration of the Alertmanager.
+
+        Uses Alertmanager's API to get the current configuration.
+
+        Returns:
+            str: Alertmanager configuration
+        """
+        config_endpoint = "api/v2/status"
+        url = "{}/{}".format(self.api_address, config_endpoint)
+        response = requests.get(url)
+        return response.json()["config"]["original"] if response.status_code == 200 else ""
+
+
+class AlertmanagerRemoteConfigurationConsumer(Object):
+    """API that manages a required `alertmanager_remote_configuration` relation.
+
+    The `AlertmanagerRemoteConfigurationConsumer` is intended to be used by charms that need
+    to push data to other charms over the `alertmanager_remote_configuration` interface.
+
+    The `AlertmanagerRemoteConfigurationConsumer` object can be instantiated as follows
+    in your charm:
+
+    ```
+    from charms.alertmanager_k8s.v0.alertmanager_remote_configuration import
+        AlertmanagerRemoteConfigurationConsumer,
+    )
+
+    def __init__(self, *args):
+        ...
+        self.remote_configuration_consumer = AlertmanagerRemoteConfigurationConsumer(self)
```

Comment:
Just realized that I was thinking of the consumer is alertmanager side. Any particular reason why alertmanager was chosen to be the provider side?

Reply:
As mentioned in the comment above - it's also confusing for my team ;) It all depends on the angle - is the `provider` the one that provides the configuration or the one that provides the relation?
I followed the schema from `prometheus_remote_write`, in which `Provider` is the `prometheus-k8s` and the `Consumer` is any charm that needs to use `remote_write`. 
Also @ghislainbourgeois asked me why `Consumer` and not `Requirer`? While I like `Requirer` better, it still leaves some freedom of interpretation of which side is which.
Summarizing - the reason is the `prometheus_remote_write`, but I'm open to changing that if the common understanding of roles is opposite.